server/src/signin-button.js

```diff
@@ -0,0 +1,50 @@
+const React = require("react");
+const PropTypes = require("prop-types");
+const { Localized } = require("fluent-react/compat");
+const sendEvent = require("./browser-send-event.js");
+
+exports.SignInButton = class SignInButton extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      displaySettings: props.isAuthenticated
+    };
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    return { displaySettings: nextProps.isAuthenticated };
+  }
+
+  render() {
+    if (this.state.displaySettings) {
+      return <Localized id="settingsButton">
+        <a className="settings-button" onClick={ this.clickHandler.bind(this) } aria-label="Settings">
+          <Localized id="gSettings">
+            <span>Settings</span>
+          </Localized>
+        </a>
+      </Localized>;
+    }
+    return <Localized id="signInButton">
+      <a className="signin-button" onClick={ this.clickHandler.bind(this) } aria-label="Sign In">
+        <Localized id="gSignIn">
+          <span>Sign In</span>
+        </Localized>
+      </a>
+    </Localized>;
+  }
+
+  clickHandler(event) {
+    if (this.props.isAuthenticated) {
+      window.location = "/settings";
```

Comment:
We don't really need when the correct markup is rendered for Settings.

Reply:
Updated with squashed commits